standup/auth0/settings.py

```diff
@@ -0,0 +1,157 @@
+                &redirect_uri={AUTH0_CALLBACK_URL}&scope=openid+email+profile&response_type=code
+
+            Look at the Auth0 and Mozilla IAM documentation for how to set up the url.
+            """)
+        ),
+        Option(
+            'AUTH0_CALLBACK_URL',
+            help_text=dedent("""\
+            This is the complete url to the callback view of your app that Auth0
+            will redirect to after the user has authenticated.
+
+            It needs to be something like ``https://yourdomain/auth/login``.
+            """)
+        ),
+        Option(
+            'AUTH0_SIGNIN_VIEW',
+            help_text=dedent("""\
+            The Django view name for the view of your signin page.
+
+            For example, on Standup, we have a separate page for signing in and the
+            Django view name is ``users.loginform``.
+
+            This is used every time we need to logout a user and redirect them to a
+            signin form.
+            """)
+        ),
+        Option(
+            'AUTH0_PATIENCE_TIMEOUT', default=5, parser=int,
+            help_text=dedent("""\
+            The amount of time in seconds that your app will wait when
+            sending requests to the Auth0 server.
+            """)
+        ),
+        Option(
+            'AUTH0_ID_TOKEN_DOMAINS',
+            help_text=dedent("""\
+            list of strings: The domains that require an ``id_token`` for your app.
+
+            For Mozilla sites, this is something like::
+
+                ['mozilla.com', 'mozillafoundation.org', 'mozilla-japan.org']
+
+            If someone logs into Auth0 using an account that has a verified email address
+            with one of those domains, then they'll be required to have logged in using
+            an Oauth2 Auth0 provider like the Mozilla LDAP option. If they didn't use
+            such a provider, they're immediately logged out and told to use an
+            appropriate provider.
+            """)
+        ),
+        Option(
+            'AUTH0_ID_TOKEN_EXPIRY', default=900, parser=int,
+            help_text=dedent("""\
+            Users who used an Oauth2 Auth0 provider and have a verified
+            email address in ``AUTH0_ID_TOKEN_DOMAINS`` have their ``id_token`` renewed
+            every ``AUTH0_ID_TOKEN_EXPIRY`` seconds.
+
+            If the ``id_token`` fails renewal, the user is immediately logged out.
+            """)
+        )
+    ]
+
+    def __init__(self):
+        self._option_map = OrderedDict([
+            (opt.key, opt) for opt in self._all_options
+        ])
+
+    def get(self, key):
+        opt = self._option_map[key]
+        val = getattr(settings, opt.key, NO_VALUE)
+        if val is not NO_VALUE:
+            return opt.parser(val)
+
+        if opt.default is not NO_VALUE:
+            return opt.parser(opt.default)
+        raise ConfigurationError('"%s" must be configured in settings. See docs.' % opt.key)
```

Comment:
Mmm... We can tweak this message and include the `help_text`. Plus the `opt.parser` calls should be wrapped in try/except with similar error handling.